src/utils.cpp

```diff
@@ -1176,18 +1176,21 @@ std::string utils::get_basename(const std::string& url)
 	char* base = NULL;
 	xmlURIPtr uri = xmlParseURI(buf);
 	if (uri && uri->path) {
-		base = basename(uri->path);
-		if (base) {
-			// check for empty path
-			if (base[0] == '/') {
-				base = NULL;
+		std::string path(uri->path);
+		// check for path ending with an empty filename
+		if (path[path.length() - 1] != '/') {
+			base = basename(uri->path);
+			if (base) {
+				// check for empty path
+				if (base[0] == '/') {
+					base = NULL;
```

Comment:
Since `base` is not used after this `if`, it doesn't make sense to assign to it anymore. I suggest flipping the condition to `base[0] != '/'` and keeping just the code from the `else` branch.